src/Verilator.cpp

```diff
@@ -139,6 +140,15 @@ static void process() {
     V3Dead::deadifyModules(v3Global.rootp());
     v3Global.checkTree();
 
+    // Create a hierarchical verilation plan
+    if (!v3Global.opt.lintOnly() && !v3Global.opt.xmlOnly() && v3Global.opt.hierarchical()) {
+        V3HierBlockPlan::createPlan(v3Global.rootp());
+        // If a plan is created, further analysis is not necessary.
+        // The actual Verilation will be done based on this plan.
+        if (v3Global.hierPlanp()) return;
+    }
+    V3Stats::addStat("HierBlock, Hierarchy blocks", v3Global.opt.hierBlocks().size());
```

Comment:
Perhaps this should be moved to inside createPlan?

Reply:
Moved to createPlan() in 2c3cfbf0692fc2e7e4673d2bd1853eb994f958c6